### PR TITLE
Sheffield | 26-Jan-ITP | Daniel Aderibigbe | Sprint 1 | Coursework/sprint 1

### PR DESCRIPTION
**Pull request by Dan2Clouted**
## Learners, PR Template

Self checklist

- [x] I have titled my PR with Region | Cohort | FirstName LastName | Sprint | Assignment Title
- [x] My changes meet the requirements of the task
- [x] I have tested my changes
- [x] My changes follow the [style guide](https://curriculum.codeyourfuture.io/guides/reviewing/style-guide/)

## Changelist

Completed all exercises.

### CODE REVIEW

File: Sprint-1/implement/dedupe.js [hunk 1]
```diff
@@ -1 +1,12 @@
-function dedupe() {}
+function dedupe(array) {
+  const result = [];
+
+  for (let i = 0; i < array.length; i++)
+    if (!result.includes(array[i])) {
+      result.push(array[i]);
+    }
+
```
Comment (cjyuan) [inline Sprint-1/implement/dedupe.js]: - Better practice is to always use `{ ... }` even if the for loop is controlling only one statement.

- When the code in a for-loop does not need array index, a for-of loop is is usually more readable.

**Comment by Dan2Clouted** [comment]
Thanks for the detailed feedback — really helpful.

I’ve gone through all the comments and made the following updates:

* Refactored loops (e.g. using `for...of` where appropriate and adding braces for clarity)
* Improved tests to ensure correctness, including checking that returned arrays are new copies rather than the original reference
* Handled edge cases in functions like `max` and `sum`, including ignoring `NaN` and non-finite values
* Updated decimal tests to use `toBeCloseTo` to avoid floating point precision issues
* Removed duplicate tests and improved test coverage where needed

**Review by cjyuan — COMMENTED** [review]
Changes look good, but you introduced a bug into one of the files. Can you fix it?

**Comment by Dan2Clouted** [comment]
The bug issues has been resolved.

**Comment by cjyuan** [comment]
I can still see the bug. If you run jest test, you should notice some failed tests.

**Comment by Dan2Clouted** [comment]
I think now it should have been fixed.

**Comment by cjyuan** [comment]
No new commits pushed to GitHub, and label not yet changed.

**Comment by Dan2Clouted** [comment]
Sorry for all this, But i think it should be fixed now. Thank you for your time.

**Comment by cjyuan** [comment]
One of the commit fixed the bug but it wasn't pushed to GitHub before. However, a newer commit reintroduced the bug.

So the bug remains.

**Comment by Dan2Clouted** [comment]
It should be fixed now. Thank you for your time and patience.

**Comment by illicitonion** [comment]
Closing PR because the January ITP run has finished. Feel free to re-open if you're still working on it.
